### PR TITLE
Check return value from query_auxv before using it

### PR DESCRIPTION
**Pull request by DanHeidinga**
There's a crash on PPC due to assuming query_auxv never fails.
The return value needs to be checked for -1 or 0 before using it
to map to something.

fixes: #1476

Signed-off-by: Dan Heidinga <daniel_heidinga@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
jenkins test sanity plinux

**Comment by DanHeidinga** [comment]
For debugging auxv issue, I found this useful:
```
 $ LD_SHOW_AUXV=1 sleep 1
```
as it prints the AUXV output. See http://man7.org/linux/man-pages/man3/getauxval.3.html for details

**Comment by DanHeidinga** [comment]
Updated the commit as the original just check the same variable over and over.

Behaviour is now:

We can't check the AT_HWCAP queries as they may actually return
0 or -1 as the field being looked up is a bit field. Assuming
they succeed is the best we can do given the API.

Allow the AT_BASE_PLATFORM call to fail and default to UNKNOWN
if the AT_PLATFORM has already succeeded.

**Comment by DanHeidinga** [comment]
jenkins test sanity plinux

File: runtime/port/unix/j9sysinfo.c [hunk 1]
```diff
@@ -479,20 +479,34 @@ getLinuxPPCDescription(struct J9PortLibrary *portLibrary, J9ProcessorDesc *desc)
...
 		desc->physicalProcessor = PROCESSOR_PPC_UNKNOWN;
+		desc->features[0] = 0;
+		desc->features[1] = 0;
 		return -1;
 	}
 
 	/* Linux PPC processor */
 	platform = (char *) query_auxv(AT_PLATFORM);
+	if ((NULL == platform) || (((char *) -1) == platform)) {
+		goto _error;
```
Comment (zl-wang) [inline runtime/port/unix/j9sysinfo.c]: This goto is grotesque.  Why not initialize outside the info and avoid it all together?

Comment (DanHeidinga) [reply runtime/port/unix/j9sysinfo.c]: :)  goto is used this way throughout the VM code base to simplify error handling.  

If you prefer the code be written another way, please open a PR.

**Pull request merged**